lldb/source/Plugins/TypeSystem/Swift/SwiftASTContext.cpp

```diff
@@ -1294,6 +1302,29 @@ static bool DeserializeAllCompilerFlags(swift::CompilerInvocation &invocation,
                 << getImportFailureString(result) << "\n";
           return false;
         }
+
+        // In case of explicit modules, for now fallback to implicit
+        // module loading.
+        // TODO: Incorporate loading explicit module dependencies to
+        // speedup dependency resolution.
+        if (llvm::find(invocation.getExtraClangArgs(), "-fno-implicit-modules")) {
```

Comment:
Out of curiosity, would `SwiftASTContext::AddExtraClangArgs` be a more canonical place to filter this out, since we already do filtering there?(compare to https://github.com/apple/llvm-project/pull/8025)

Reply:
Are you still planning to do this, or is it not a good idea for reasons? (Doing it in a follow-up is fine)

Reply:
I am going to take a look at moving this to `AddExtraClangArgs` now, thanks.

Reply:
Updated the commit to filter in `SwiftASTContext::AddExtraClangArgs`, thanks @adrian-prantl .